sdks/python/apache_beam/runners/worker/opcounters.py

```diff
@@ -219,9 +219,14 @@ def update_from_batch(self, windowed_batch):
     assert self.producer_batch_converter is not None
     assert isinstance(windowed_batch, windowed_value.HomogeneousWindowedBatch)
 
-    self.element_counter.update(
-        self.producer_batch_converter.get_length(windowed_batch.values))
-    # TODO(BEAM-14408): Update byte size estimate
+    batch_length = self.producer_batch_converter.get_length(
+        windowed_batch.values)
+    self.element_counter.update(batch_length)
+
+    mean_element_size = self.producer_batch_converter.estimate_byte_size(
+        windowed_batch.values) / batch_length
+    for _ in range(batch_length):
+      self.mean_byte_counter.update(mean_element_size)
```

Comment:
This seems very naive, but I'm not sure of another way to keep this consistent with the element-wise behaviour, which is tracking the distribution of _element_ byte size estimates. Definitely open to suggestions here.

Reply:
Look at adding an `update_n` to distribution counter